pkg/cni/install.go

```diff
@@ -76,6 +82,43 @@ func (i *Installer) Start() error {
 			i.Stop()
 			return err
 		}
+
+		watcher, err := fsnotify.NewWatcher()
+		if err != nil {
+			return fmt.Errorf("failed to create fsnotify watcher: %v", err)
+		}
+		i.Watcher = watcher
+
+		if err = watcher.Add(ServiceAccountPath); err != nil {
+			return fmt.Errorf("failed to add fsnotify watcher for path %s: %v", ServiceAccountPath, err)
+		}
+
+		// Start listening for events.
+		go func() {
+			log.Infof("start watching directory %s", ServiceAccountPath)
+
+			for {
+				select {
+				case _, ok := <-watcher.Events:
+					if !ok {
+						log.Info("events channel of fsnotify watcher was closed")
+						return
+					}
+
+					// NOTE: due to the update mechanism of service account path, try to update
+					// kubeconfig of Kmesh cni regardless of any events.
+					if err := maybeWriteKubeConfigFile(filepath.Join(i.CniMountNetEtcDIR, kmeshCniKubeConfig)); err != nil {
```

Comment:
This can be improved, when sa changed, there can be multi events received. We should only run rewrite once.